src/openvic-simulation/pop/Pop.cpp

```diff
@@ -162,6 +172,59 @@ void Pop::update_gamestate(
 //TODO store income
 void Pop::add_rgo_owner_income(const fixed_point_t income) {}
 void Pop::add_rgo_worker_income(const fixed_point_t income) {}
+//adds income to the temporary stores and then to general income store
+void Pop::add_artisanal_income(const fixed_point_t income){
+	artisanal_income += income;
+	this->income += income;
```

Comment:
In all of these income adding functions, change the argument's name to something other than `income` so there's no risk of clash with the `Pop`'s `income` member variable, and then remove all the `this->`s which will no longer be necessary

Reply:
`amount` for example